File: examples/close-session/rest/java/CloseSession.java
// Creates a Browserless session and closes it via the Session API.
//
// Run: javac CloseSession.java && java CloseSession

import java.net.URI;
import java.net.http.*;

public class CloseSession {
    public static void main(String[] args) throws Exception {
        String token = "YOUR_API_TOKEN_HERE";
        String endpoint = "https://production-sfo.browserless.io/session?token=" + token;

        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(endpoint))
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(
                "{\"ttl\": 60000, \"stealth\": true}"
            ))
            .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        String body = response.body();
        System.out.println("Session created: " + body);

        // Parse stop URL from response
        int stopStart = body.indexOf("\"stop\":\"") + 8;
        int stopEnd = body.indexOf("\"", stopStart);
        String stopUrl = body.substring(stopStart, stopEnd);

        HttpRequest deleteRequest = HttpRequest.newBuilder()
            .uri(URI.create(stopUrl + "&force=true"))
            .DELETE()
            .build();

        HttpResponse<String> deleteResponse = client.send(
            deleteRequest, HttpResponse.BodyHandlers.ofString()
        );
        System.out.println("Session closed: " + deleteResponse.statusCode());
    }
}

File: examples/close-session/rest/csharp/CloseSession.cs
// Creates a Browserless session and closes it via the Session API.
//
// Run: dotnet run

using System.Net.Http;
using System.Text;
using System.Text.Json;

var token = "YOUR_API_TOKEN_HERE";
var client = new HttpClient();

var content = new StringContent(
    JsonSerializer.Serialize(new { ttl = 60000, stealth = true }),
    Encoding.UTF8,
    "application/json"
);

var response = await client.PostAsync(
    $"https://production-sfo.browserless.io/session?token={token}",
    content
);
var body = await response.Content.ReadAsStringAsync();
var session = JsonSerializer.Deserialize<JsonElement>(body);

var stopUrl = session.GetProperty("stop").GetString();
Console.WriteLine($"Session created: {session.GetProperty("id")}");

var closeResponse = await client.DeleteAsync($"{stopUrl}&force=true");
Console.WriteLine($"Session closed: {closeResponse.StatusCode}");
